engine/src/exec-interface-stack.cpp

```diff
@@ -2182,3 +2183,29 @@ void MCStack::SetScriptOnly(MCExecContext& ctxt, bool p_script_only)
 {
     m_is_script_only = p_script_only;
 }
+
+// MERG-2015-10-11: [[ DocumentFilename ]] Add stack documentFilename property
+void MCStack::GetDocumentFilename(MCExecContext &ctxt, MCStringRef& r_document_filename)
+{
+    if (m_document_filename == nil)
+        return;
+    
+    r_document_filename = MCValueRetain(m_document_filename);
+}
+
+void MCStack::SetDocumentFilename(MCExecContext &ctxt, MCStringRef p_document_filename)
+{
+    MCStringRef t_resolved_filename;
+    
+    if (!MCS_resolvepath(p_document_filename, t_resolved_filename))
```

Comment:
Ah, yes I just noticed that it will now end up with the icon for the default folder... good point. Will fix